lambda/repository/lambda_functions.py

```diff
@@ -545,12 +1116,106 @@ def create(event: dict, context: dict) -> Any:
     return {"status": "success", "executionArn": response["executionArn"]}
 
 
+@api_wrapper
+def get_repository_by_id(event: dict, context: dict) -> Dict[str, Any]:
+    """
+    Get a vector store configuration by ID.
+
+    Args:
+        event (dict): The Lambda event object containing:
+            - pathParameters.repositoryId: The repository ID to retrieve
+        context (dict): The Lambda context object
+
+    Returns:
+        Dict[str, Any]: The repository configuration with default values for new fields
+
+    Raises:
+        ValidationError: If repositoryId is missing
+        HTTPException: If repository not found or access denied
+    """
+    # Extract path parameters
+    path_params = event.get("pathParameters", {})
+    repository_id = path_params.get("repositoryId")
+
+    if not repository_id:
+        raise ValidationError("repositoryId is required")
+
+    # Get repository and check access
+    repository = get_repository(event, repository_id)
+
+    return repository
+
+
+@api_wrapper
+@admin_only
+def update_repository(event: dict, context: dict) -> Dict[str, Any]:
+    """
+    Update a vector store configuration. This function is only accessible by administrators.
+
+    Args:
+        event (dict): The Lambda event object containing:
+            - pathParameters.repositoryId: The repository ID to update
+            - body: JSON with fields to update (UpdateVectorStoreRequest)
+        context (dict): The Lambda context object
+
+    Returns:
+        Dict[str, Any]: The updated repository configuration
+
+    Raises:
+        ValidationError: If validation fails
+        HTTPException: If repository not found
+    """
+    from models.domain_objects import UpdateVectorStoreRequest
+
+    # Extract path parameters
+    path_params = event.get("pathParameters", {})
+    repository_id = path_params.get("repositoryId")
+
+    if not repository_id:
+        raise ValidationError("repositoryId is required")
+
+    # Parse request body
+    try:
+        body = json.loads(event.get("body", {}))
+        request = UpdateVectorStoreRequest(**body)
+    except json.JSONDecodeError as e:
+        raise ValidationError(f"Invalid JSON in request body: {e}")
+    except Exception as e:
+        raise ValidationError(f"Invalid request: {e}")
+
+    # Ensure repository exists
+    _ = vs_repo.find_repository_by_id(repository_id)
+
+    # Build updates dictionary (only include fields that were provided)
+    updates = {}
+    if request.repositoryName is not None:
```

Comment:
These wouldn't be booleans so the more idiomatic `if request.repositoryName:` is probably more appropriate and much easier to read in this dense chain of conditions.